src/rgw/rgw_rados.cc

```diff
@@ -7542,7 +7548,11 @@ int RGWRados::raw_obj_stat(rgw_raw_obj& obj, uint64_t *psize, real_time *pmtime,
     op.stat2(&size, &mtime_ts, NULL);
   }
   if (first_chunk) {
-    op.read(0, cct->_conf->rgw_max_chunk_size, first_chunk, NULL);
+    if (range == nullptr) {
+      op.read(0, cct->_conf->rgw_max_chunk_size, first_chunk, NULL);
```

Comment:
`first_chunk` and `range` go together; we shouldn't ever have one without the other, right? lets skip this part, and just call read `if (first_chunk && range) {`